riot/lol/match.go

```diff
@@ -78,29 +67,35 @@ func (m *MatchClient) List(accountID string, beginIndex, endIndex int, options .
 
 // MatchStreamValue value returned by ListStream, containing either a reference to a match or an error
 type MatchStreamValue struct {
-	*MatchReference
-	Error error
+	MatchID string
+	Error   error
 }
 
 // ListStream returns all matches played on this account as a stream, requesting new until there are no
 // more new games
-func (m *MatchClient) ListStream(accountID string) <-chan MatchStreamValue {
+func (m *MatchClient) ListStream(puuid string, options ...*MatchListOptions) <-chan MatchStreamValue {
 	logger := m.logger().WithField("method", "ListStream")
 	cMatches := make(chan MatchStreamValue, 100)
+
+	if len(options) != 0 && options[0].Queue != nil {
+		// Copy the value in case the caller modifies it after we return
+		queue := *options[0].Queue
+		options[0].Queue = &queue
+	}
 	go func() {
+		defer close(cMatches)
```

Comment:
Channels should always be closed after the last write. Also, the act of closing a channel is sufficient to signal an end of data, so no need to send an EOF error